### ISSUE TITLE
Are _label properties supposed to be expanded into rdfs:label triples?

### ISSUE THREAD

**Issue by bluebinary** [linked-art/linked.art#539]
If I recall early Linked.Art lore correctly, I believe the intention of renaming the `label` property (as it was known originally) to `_label` was to more strongly indicate its sole purpose as a simple note to help developers or others reading the raw JSON-LD navigate and contextualise the JSON, lessening the need to dereference links to gain a clear sense of what they referred to.

I believe we have a common – and likely documented – understanding that the `_label` property is not meant to be used programmatically or relied upon in code or for user experiences.

Furthermore, I believe the _underscore_ naming-convention was used to note that the property would not be expanded into triples through a standard graph expansion process. However, as the `_label` property currently maps to `rdfs:label` via the current version of the Linked.Art context document, `_label` properties are currently expanded into triples unless one overrides default graph expansion behaviour to prevent it.

One of the issues the current graph expansion behaviour causes is if there are inconsistencies between `_label` values used for the same `id` across a set of documents in the same graph store, such as for classifications where the `_label` values may unintentionally or intentionally vary (for clarity). This can lead to multiple `rdfs:label` tripes being associated with the same node. Their existence in the graph store also seems counter to their purpose as notes/labels for navigating and contextualising the raw JSON-LD.

As such the question is: do we actually want `_label` properties to continue being expanded into graph tripes?

If so, should we possibly reconsider which ontology they map to? Do we wish to continue mapping `_label` to `rdfs:label ` or would `dcterms:label` for example be a more appropriate mapping?

If not, should we not remove the `_label` property to `rdfs:label` mapping from the context document for the Linked.Art `1.0` release to prevent future graph expansion of `_label` properties?

**Comment by azaroth42** [comment]
You are correct in everything above! `_label` currently expands to rdfs:label, shouldn't be used in UIs or its value relied on programatically, it's a hint to some human reading the JSON as to what the thing is that has the URI in `id`. Underscore is there to visually separate it out.

The rationale for having it expand to `rdfs:label` was for vanilla UIs on top of the RDF data that look for `rdfs:label` and render it as the title of the view. The wikidata pages are a (slightly) friendlier version of that tool, which someone can likely recall the name of. Again, no end user should see that rendering either but perhaps it's no longer necessary and we should assume that there will be tools that understand how to get the primary name out of the linked art structure?

If so ... then I think we should remove the mapping from the context document and let `_` be our bridge to the API and not semantic at all.

**Comment by azaroth42** [comment]
Further thinking on this ...

If the data is being generated from a triplestore, then not having `_label` mapped in the context would make compaction impossible. There would need to be a second layer that annotated each JSON object in the result with its `_label`, likely by doing a search for the primary name of the entity and embedding it. However ... if it's an internal node like a `TimeSpan`, `Identifier`, or `Dimension` then there would be nowhere to look for this information.

So in retrospect, I think I'd leave it how it is.

@edwardanderson and others who are building from triplestores ... your thoughts?

**Comment by cbutcosk** [comment]
As a graph user another issue is on the consumption side, where the JSON-LD -> RDF quads/triples expansion is quite useful (eg, for deconstructing to drawing-list forms like mermaid) and removing labels from the quad/triples representation is not quite the same thing as adding an underscore to the property name in JSON.

**Comment by azaroth42** [comment]
@beaudet @edwardanderson @krpage Thoughts?

**Comment by beaudet** [comment]
I'm a big fan of being able to easily round-trip a Linked Art JSON graph to RDF and back without losing fidelity or completeness and for that reason alone, I think it should remain mapped.  It's easy enough to filter out _label if an implementation doesn't want it in other representations.  Related to #461

**Comment by edwardanderson** [comment]
> This can lead to multiple rdfs:label triples being associated with the same node.

We have exactly this problem when creating our Linked Art triples from a store containing a mixture of different models and an inconsistent amount of `rdfs:label` (or other) literals. It's primarily a challenge for us to generate a single `_label` value since we're striving to perform one-shot transformation for each entity type in a single `SPARQL CONSTRUCT` query and haven't figured out how to constrain the number of instances of a particular triple pattern to `1` (related to [this "distributed limit" problem](https://github.com/w3c/sparql-dev/issues/100) I think).

We could likely solve this either with some post-processing to remove extraneous labels (maybe inconsistently?) or building some kind of [Wikidata-style `SERVICE` solution for labelling](https://en.wikibooks.org/wiki/SPARQL/SERVICE_-_Label). But for now we are simply omitting the labels altogether.

More generally I don't have an opinion whether `_label` should be semantic or not, but I feel like the work on the [Related Entitles API](https://github.com/linked-art/linked.art/issues/412) is starting to generate other `_`-prefixed API properties which explicitly aren't semantic... and maybe it's good to be consistent across them all? Equally, round-tripping the data without having to pre-process it is nice and handy.

**Comment by azaroth42** [comment]
Should roll into #461 discussion. Leave `_label` as is, pending a resolution for #461 (if there is one!)
